src/main/java/com/yahoo/bullet/record/simple/TypedSimpleBulletRecord.java

```diff
@@ -56,6 +57,11 @@ public TypedSimpleBulletRecord remove(String field) {
         return this;
     }
 
+    @Override
+    public TypedObject typedGet(String field, Type hint) {
```

Comment:
So for this, the hint is basically ignored. Do we want to go the route of the hint takes priority or the data takes priority. Here we're doing data. Other classes do hint.

Reply:
My intention is for the hint to be used similar to the default in a getOrDefault(...). If the type is already known, then use that. If the type must be "figured out" through TypedObject instantiation, then the type was not previously known so we use the hint in that case. I think I've been consistent with that idea.